File: app/src/main/java/com/felipemiranda/desafioconcrete/ui/main/BaseViewHolder.java
package com.felipemiranda.desafioconcrete.ui.main;

import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

/**
 * Created by felipemiranda
 */

public abstract class BaseViewHolder extends RecyclerView.ViewHolder {
    private int mCurrentPosition;
    public BaseViewHolder(View itemView) {
        super(itemView);
    }
    protected abstract void clear();
    public void onBind(int position) {
        mCurrentPosition = position;
        clear();
    }
    public int getCurrentPosition() {
        return mCurrentPosition;
    }
}

File: app/src/main/java/com/pedrenrique/githubapp/features/common/adapter/BaseViewHolder.kt
package com.pedrenrique.githubapp.features.common.adapter

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.pedrenrique.githubapp.features.common.adapter.factory.TypesFactory

abstract class BaseViewHolder<T>(view: View) : RecyclerView.ViewHolder(view) {
    abstract fun bind(item: T, typesFactory: TypesFactory)
}